src/blocks/net.rs

```diff
@@ -936,3 +908,150 @@ impl Block for Net {
         &self.id
     }
 }
+
+fn get_ssid(dev: &NetworkDevice) -> Result<Option<String>> {
+    if let Some(res) = get_iw_ssid(dev)? {
+        return Ok(Some(res));
+    }
+
+    if let Some(res) = get_wpa_ssid(dev)? {
+        return Ok(Some(res));
+    }
+
+    if let Some(res) = get_nmcli_ssid(dev)? {
+        return Ok(Some(res));
+    }
+
+    if let Some(res) = get_iwctl_ssid(dev)? {
+        return Ok(Some(res));
+    }
+
+    Ok(None)
+}
+
+#[inline]
+/// Attempt to get the SSID the given device is connected to from iw.
+/// Returns Err if:
+///     - `iw` is not a valid command
+///     - failed to spawn an `iw` command
+///     - `iw` failed to produce a valid UTF-8 SSID
+/// Returns Ok(None) if `iw` failed to produce a SSID.
+fn get_iw_ssid(dev: &NetworkDevice) -> Result<Option<String>> {
+    let raw = Command::new("iw")
```

Comment:
Do you think it's worth DRYing this block that's also used in the other get_*_ssid functions?

Reply:
Which block? The lines shown are the start of a function.
Those functions are quite repetitive in general, so pulling some of it out into a helper function might be a good idea.

Reply:
Sorry, was on mobile.

This part that's common across the different functions:
 ```
let raw... 
... 
return Err(err).block_error("net", "Failed to execute SSID query using xx.");
```

Reply:
You can leave it for another time if you want

Reply:
No worries, it does make sense to pull that section out.